### PR TITLE
Enhancement: more logging in E2E tests + turn back on commit-workflow E2E test

### PR DESCRIPTION
**Pull request by tzaffi**
### Original explanation:

> Pass the nightly tests by updating imported `type.EvalDelta` from `go-algorand-sdk` and updating `conduit` dependency. 

This no longer applies, as tests are passing because the conduit dependency has been updated.
However, on-commit E2E tests were commented out in a previous PR, and this turns them back on.

Additional changes:
* update pinned AWS E2E key to latest available
* add `env` printout and other breadcrumbs in e2e `Dockerfiles`
* refactor python scripts to make them a little easier to work with/debug

## Issue

#1521 

## Related PR's
* go-algorand-sdk PR: https://github.com/algorand/go-algorand-sdk/pull/531
* conduit PR: https://github.com/algorand/conduit/pull/73
* go-algorand PR: https://github.com/algorand/go-algorand/pull/5381 for cross-repo type checking

### CODE REVIEW

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [inline e2e_tests/src/e2e_indexer/e2elive.py:115]: TODO: don't merge with this commented out (revert or comment out)

File: e2e_tests/docker/indexer/Dockerfile
Comment (tzaffi) [inline e2e_tests/docker/indexer/Dockerfile:25]: TODO: don't merge with this branch

```suggestion
RUN git clone -b master https://github.com/algorand/conduit
```

File: .circleci/config.yml
Comment (tzaffi) [inline .circleci/config.yml:57]: `run_e2e_tests` is still failing

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [inline e2e_tests/src/e2e_indexer/e2elive.py:47]: in `--verbose` mode, log breadcrumbs such as:
```
DEBUG:e2e_indexer.e2elive:<<<<<Entering xrun>>>>>
```

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [inline e2e_tests/src/e2e_indexer/e2elive.py:116]: run `validate_accounting()` directly, instead of in a separate process.

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [inline e2e_tests/src/e2e_indexer/e2elive.py:377]: these types are called too often and create too much noise in the logs

File: e2e_tests/src/e2e_indexer/validate_accounting.py
Comment (tzaffi) [inline e2e_tests/src/e2e_indexer/validate_accounting.py:45]: Refactoring so can call directly via `nxrun()`

File: e2e_tests/src/e2e_indexer/validate_accounting.py
Comment (tzaffi) [inline e2e_tests/src/e2e_indexer/validate_accounting.py:760]: Strange behavior for failing tests - failing while trying to rewind an account with an application transaction. Seems like this issue would be present regardless of the difference between `rel-nightly` and `stable`.

File: go.mod
Comment (tzaffi) [inline go.mod:7]: TODO: Do not merge as is ^^^. Only merge after merging https://github.com/algorand/go-algorand-sdk/pull/531 and updating with `go get go-algorand-sdk`.

File: e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile
Comment (tzaffi) [inline e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile:28]: TODO: don't merge with this branch

```suggestion
RUN git clone -b master https://github.com/algorand/conduit
```

File: .circleci/config.yml
Comment (Eric-Warehime) [reply .circleci/config.yml:57]: Do you have the output from the failure? I only ask because the only differences here should be that we grab the algod binary from the stable channel instead of nightly, and all of the changes we need for them to work should be in stable.

File: e2e_tests/src/e2e_indexer/validate_accounting.py
Comment (Eric-Warehime) [reply e2e_tests/src/e2e_indexer/validate_accounting.py:760]: I remember something similar happening when I was working on these tests before. I think it was just from a mismatch between rounds...i.e. we're not successfully importing in the way we should be, or there was a failure happening that was not evident without looking into some specific logs. Sorry if that's not super helpful.

Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/validate_accounting.py:760]: (not resolving to show the original comment that I'm about to remove)

File: .circleci/config.yml
Comment (tzaffi) [reply .circleci/config.yml:57]: failed to retrieve `StateDelta`. Here's the failing test on a draft PR: https://app.circleci.com/pipelines/github/algorand/indexer/3802/workflows/e15022f5-a6a5-4911-95a6-19f03949d029/jobs/5345

Comment (shiqizng) [reply .circleci/config.yml:57]: I believe it's a bug. There is no deltas for round 0. We shouldn't be making that request, https://github.com/algorand/conduit/blob/master/conduit/plugins/importers/algod/algod_importer.go#L234

Comment (tzaffi) [reply .circleci/config.yml:57]: Thanks for pointing out the bug @shiqizng. I have a PR out to fix it: https://github.com/algorand/conduit/pull/78

Unfortunately, that was a red herring. We can see that after I linked my E2E indexer test branch to the conduit branch with the fix, I still fail the test but this time with even less information:
```sh
indexer-e2e-1     | ERROR:e2e_indexer.validate_accounting:failed to fetch 'http://localhost:9890/v2/accounts/BAKLOSILWYXN2ZYNP77KIF42LWTO3XM6SIXJEVRMQB64PFBGG3AGF2OAVM?round=31'
```

When I run it locally verbose, the first detailed **ERROR:** I get is the following:

```sh
indexer-e2e-1     | ERROR:e2e_indexer.validate_accounting:failed to fetch 'http://localhost:9890/v2/accounts/BAKLOSILWYXN2ZYNP77KIF42LWTO3XM6SIXJEVRMQB64PFBGG3AGF2OAVM?round=31'
```

Comment (tzaffi) [reply .circleci/config.yml:57]: I believe the original comment here still applies so that once the current `rel-nightly' is released the test will pass.

**Comment by tzaffi** [comment]
> Looks mostly good once we resolve some of the outstanding todos.

I believe I've resolved these TODO's except for the long-standing issue that we discussed above.

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (shiqizng) [inline e2e_tests/src/e2e_indexer/e2elive.py:111]: why don't we need the timeout anymore?

File: e2e_tests/docker/indexer/Dockerfile
Comment (bbroder-algo) [inline e2e_tests/docker/indexer/Dockerfile:25]: Hmm...

File: e2e_tests/src/e2e_common/util.py
Comment (bbroder-algo) [inline e2e_tests/src/e2e_common/util.py:127]:
```suggestion
    gnxrun = Gnxrun Not Xrun
```

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (bbroder-algo) [inline e2e_tests/src/e2e_indexer/e2elive.py:163]: I'm not so sure this timeout was even guaranteed to engage, see https://stackoverflow.com/questions/36952245/subprocess-timeout-failure

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (bbroder-algo) [inline e2e_tests/src/e2e_indexer/e2elive.py:111]: nxrun has a idempotentcy requirement for func over xrun if run multiple times, but I assume validate_accounting meets that requirement

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (bbroder-algo) [inline e2e_tests/src/e2e_indexer/e2elive.py:361]: clever but kind of nutty?

File: e2e_tests/src/e2e_common/util.py
Comment (bbroder-algo) [inline e2e_tests/src/e2e_common/util.py:135]: I think yes its better to do it this way over using subprocess.

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/e2elive.py:163]: Added this observation as an item on #1525 
image

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/e2elive.py:111]: Not sure I see the idempotency requirement. Does `xrun` have this? If so, I'll probably want to rename `nxrun` to `run_with_exit_codes` or something like that.

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/e2elive.py:361]: Just to produce breadcrumbs without having to modify every function directly. Any objections?

File: e2e_tests/src/e2e_common/util.py
Comment (tzaffi) [reply e2e_tests/src/e2e_common/util.py:135]: for calling non-python - no such luck

File: e2e_tests/src/e2e_indexer/e2elive.py
Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/e2elive.py:111]: After convo I understand the question better. Yes, this new approach is introducing the possibility of side effects in the running python environment, while the former approach shielded the environment. In a production setting, this may be undesirable. But for the purposes of testing, I think it's okay.

Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/e2elive.py:111]: I'll add a comment to `def xrun()` alerting of this issue.

Comment (tzaffi) [reply e2e_tests/src/e2e_indexer/e2elive.py:111]: cf. 69a51075b9601ccb0385f88f9d2db4d5668a09df

File: .circleci/config.yml
Comment (tzaffi) [inline .circleci/config.yml:49]: updating to a more recent test

Comment (winder) [reply .circleci/config.yml:49]: more recent, this definitely has the new txn-counter changes
```suggestion
      CI_E2E_FILENAME: "f99e7b0c/rel-nightly"
```

File: e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile
Comment (winder) [inline e2e_tests/docker/indexer-filtered/e2e-write/Dockerfile:41]: nice :+1:

**Review by winder — APPROVED** [review]

**Review by shiqizng — APPROVED** [review]

**Pull request merged**
